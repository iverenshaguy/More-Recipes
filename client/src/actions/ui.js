import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM, CHANGE_RECIPE_FORM_STATE } from './actionTypes';

const toggleModal = modal => ({
  type: TOGGLE_MODAL,
  payload: modal || null
});


const toggleReviewForm = () => ({
  type: TOGGLE_REVIEW_FORM
});

const changeRecipeFormState = payload => ({
  type: CHANGE_RECIPE_FORM_STATE,
  payload
});

export default {
  toggleModal,
  toggleReviewForm,
  changeRecipeFormState
};
