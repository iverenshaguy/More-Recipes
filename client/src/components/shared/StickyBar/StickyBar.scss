@import '../../../assets/scss/modules';

.sticky-bar-wrapper {
  bottom: 0;
  color: $white;
  position: fixed;
  width: 100%;
}

.sticky-bar {
  background-color: $white;
  // bottom: 0;
  color: $light-green;
  // height: 6rem;
  padding: 1rem 0;
  width: 100%;

  i,
  span {
    color: $light-green;
  }
}

.home {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.user-profile {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.favorite-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.user-recipe {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.top-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.search-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.view-recipe {
  .sticky-bar {
    div {
      padding-right: 1.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 0.5em;
      }
    }
  }
}

@media only screen and (max-width: 768px) and (min-width: 576px) {
  .home {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .user-profile {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .favorite-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .user-recipe {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .top-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .search-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .view-recipe {
    .sticky-bar {
      div {
        padding-right: 4.2em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 0.2em;
        }
      }
    }
  }
}

@media only screen and (max-width: 575px) and (min-width: 450px) {
  .search-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .home {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .user-profile {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .favorite-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .user-recipe {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .top-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .view-recipe {
    .sticky-bar {
      div {
        padding-right: 2em;

        &:first-child {
          padding-left: 0.5em;
        }

        &:last-child {
          padding-right: 0.5em;
        }
      }
    }
  }
}

#result-search-form-div {
  background-color: $transparent-light;

  #search-form {
    #search {
      border: 1px solid $light-green !important;
    }

    .input-group-btn {
      button {
        background-color: $light-green !important;
        border-color: $light-green !important;
        color: $white !important;
      }
    }
  }
}