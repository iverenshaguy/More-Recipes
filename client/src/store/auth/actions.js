import {
  AUTHENTICATED,
  UNAUTHENTICATED,
  AUTHENTICATION_ERROR,
  AUTHENTICATING,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  SIGNUP_SUCCESS,
  SIGNUP_ERROR,
  CLEAR_AUTH_ERROR
} from './types';

const authenticating = () => ({
  type: AUTHENTICATING
});

const authenticationSuccess = data => ({
  type: AUTHENTICATED,
  payload: data
});

const authenticationFailure = error => ({
  type: AUTHENTICATION_ERROR,
  payload: error
});

const loginSuccess = data => ({
  type: LOGIN_SUCCESS,
  payload: data
});

const loginFailure = error => ({
  type: LOGIN_ERROR,
  payload: error
});

const signupSuccess = data => ({
  type: SIGNUP_SUCCESS,
  payload: data
});

const signupFailure = error => ({
  type: SIGNUP_ERROR,
  payload: error
});

const clearAuthError = () => ({
  type: CLEAR_AUTH_ERROR
});

const resetUser = () => ({
  type: UNAUTHENTICATED
});

export default {
  authenticating,
  authenticationSuccess,
  authenticationFailure,
  loginSuccess,
  loginFailure,
  signupSuccess,
  signupFailure,
  clearAuthError,
  resetUser
};
