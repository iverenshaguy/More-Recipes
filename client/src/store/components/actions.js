import { TOGGLE_MODAL } from './types';

const toggleModal = modal => ({
  type: TOGGLE_MODAL,
  payload: modal || ''
});

export default {
  toggleModal
};
