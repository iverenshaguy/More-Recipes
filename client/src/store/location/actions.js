import { HOME, AUTH } from './types';

const setHomeLocation = () => ({
  type: HOME
});

const setAuthLocation = () => ({
  type: AUTH
});

export default {
  setHomeLocation,
  setAuthLocation
};
