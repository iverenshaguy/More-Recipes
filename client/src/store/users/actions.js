import { AUTHENTICATED, AUTHENTICATION_ERROR } from './types';

const authenticationSuccess = () => ({
  type: AUTHENTICATED
});

const authenticationFailure = () => ({
  type: AUTHENTICATION_ERROR,
  payload: 'Invalid Email or Password'
});

export default {
  authenticationSuccess,
  authenticationFailure
};
