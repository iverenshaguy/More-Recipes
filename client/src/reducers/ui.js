import { TOGGLE_MODAL, TOGGLE_REVIEW_FORM } from '../actions/actionTypes';

const initialState = {
  modals: {
    isOpen: false,
    type: null
  },
  reviewForm: {
    isOpen: false
  }
};

export default (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_MODAL:
      return Object.assign({}, state, {
        modals: {
          isOpen: !state.modals.isOpen,
          type: action.payload || null
        }
      });
    case TOGGLE_REVIEW_FORM:
      return Object.assign({}, state, {
        reviewForm: {
          isOpen: !state.reviewForm.isOpen,
        }
      });
    default:
      return state;
  }
};
