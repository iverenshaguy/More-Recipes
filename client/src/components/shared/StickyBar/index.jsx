import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import FontAwesome from 'react-fontawesome';
import { toggleModal } from '../../../actions/ui';
import './StickyBar.scss';

/**
 * @exports
 * @extends Component
 * @class StickyBar
 * @classdesc Returns App Sticky Bar for Medium to Small Devices
*/
class StickyBar extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    currentLocation: PropTypes.string.isRequired
  }

  /**
   * @constructor
   * @memberof StickyBar
   * @returns {nothing} Returns nothing
   */
  constructor() {
    super();

    this.handleUserIconClick = this.handleUserIconClick.bind(this);
    this.handleHomeIconClick = this.handleHomeIconClick.bind(this);
    this.handleSearchIconClick = this.handleSearchIconClick.bind(this);
    this.handleFavoritesIconClick = this.handleFavoritesIconClick.bind(this);
    this.handleAddRecipeIconClick = this.handleAddRecipeIconClick.bind(this);
  }

  /**
   * @memberof StickyBar
   * @param {object} e
   * @returns {nothing} Returns nothing
   */
  handleHomeIconClick(e) {
    e.preventDefault();
    this.props.dispatch(push('/'));
  }

  /**
   * @memberof StickyBar
   * @param {object} e
   * @returns {nothing} Returns nothing
   */
  handleSearchIconClick(e) {
    e.preventDefault();
    this.props.dispatch(push('/'));
  }

  /**
   * @memberof StickyBar
   * @param {object} e
   * @returns {nothing} Returns nothing
   */
  handleFavoritesIconClick(e) {
    e.preventDefault();
    this.props.dispatch(push('/'));
  }

  /**
   * @memberof StickyBar
   * @param {object} e
   * @returns {nothing} Returns nothing
   */
  handleUserIconClick(e) {
    e.preventDefault();
    this.props.dispatch(push('/'));
  }

  /**
   * @memberof StickyBar
   * @param {object} e
   * @returns {nothing} Returns nothing
   */
  handleAddRecipeIconClick(e) {
    e.preventDefault();
    this.props.dispatch(toggleModal('addRecipe'));
  }

  /**
   * @memberof StickyBar
   * @returns {JSX} StickyBar
  */
  render() {
    const { currentLocation } = this.props;

    if (currentLocation === '/login' || currentLocation === '/signup') {
      return null;
    }

    return (
      <div className="view-recipe">
        <div className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none">
          <div className="sticky-bar text-center">
            <div className="nav-link d-inline">
              <FontAwesome name="home" tag="i" size="lg" className="home-icon" onClick={this.handleHomeIconClick} />
            </div>
            <div className="nav-link d-inline">
              <FontAwesome name="search" tag="i" size="lg" className="search-icon" onClick={this.handleSearchIconClick} />
            </div>
            <div className="nav-link d-inline">
              <FontAwesome
                name="plus"
                tag="i"
                id="add-edit-modal-icon"
                size="lg"
                onClick={this.handleAddRecipeIconClick}
              />
            </div>
            <div className="nav-link d-inline">
              <FontAwesome name="heart" tag="i" size="lg" className="favorite-recipes-icon" onClick={this.handleFavoritesIconClick} />
            </div>
            <div className="nav-link d-inline">
              <FontAwesome name="user" tag="i" size="lg" className="user-icon" onClick={this.handleUserIconClick} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentLocation: state.router.location.pathname
});

export { StickyBar as StickyBarComponent };

export default connect(mapStateToProps)(StickyBar);
