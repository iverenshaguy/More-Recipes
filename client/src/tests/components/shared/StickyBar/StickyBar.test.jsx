import React from 'react';
import { StickyBarComponent } from '../../../../components/shared/StickyBar';
import { toggleModal } from '../../../../actions/ui';

const dispatchMock = jest.fn();
const setup = (location) => {
  const wrapper = mount(<StickyBarComponent dispatch={dispatchMock} currentLocation={location} />);
  return { wrapper };
};

describe('StickyBarComponent', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { wrapper } = setup('view-recipe');

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.fa-plus')).toBeTruthy();
    expect(wrapper.find('.fa-search')).toBeTruthy();
    expect(wrapper.find('.fa-heart')).toBeTruthy();
    wrapper.unmount();
  });

  it("renders no sticky bar when current location is '/login'", () => {
    const { wrapper } = setup('/login');

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it("renders no sticky bar when current location is '/signup'", () => {
    const { wrapper } = setup('/signup');

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('calls modal when add recipe button is clicked', () => {
    const { wrapper } = setup('view-recipe');

    wrapper.find('i.fa-plus').simulate('click');
    expect(dispatchMock).toHaveBeenCalledWith(toggleModal('addRecipe'));
    wrapper.unmount();
  });

  it('calls handleHomeIconClick() when home icon is clicked', () => {
    const handleHomeIconClickSpy = jest.spyOn(StickyBarComponent.prototype, 'handleHomeIconClick');
    const { wrapper } = setup('view-recipe');

    wrapper.find('i.fa-home').simulate('click');
    expect(handleHomeIconClickSpy).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('calls handleSearchIconClick() when search icon is clicked', () => {
    const handleSearchIconClickSpy = jest.spyOn(StickyBarComponent.prototype, 'handleSearchIconClick');
    const { wrapper } = setup('view-recipe');

    wrapper.find('i.fa-search').simulate('click');
    expect(handleSearchIconClickSpy).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('calls handleFavoritesIconClick() when favorites icon is clicked', () => {
    const handleFavoritesIconClickSpy = jest.spyOn(StickyBarComponent.prototype, 'handleFavoritesIconClick');
    const { wrapper } = setup('view-recipe');

    wrapper.find('i.fa-heart').simulate('click');
    expect(handleFavoritesIconClickSpy).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('calls handleUserIconClick() when user icon is clicked', () => {
    const handleUserIconClickSpy = jest.spyOn(StickyBarComponent.prototype, 'handleUserIconClick');
    const { wrapper } = setup('view-recipe');

    wrapper.find('i.fa-user').simulate('click');
    expect(handleUserIconClickSpy).toHaveBeenCalled();
    wrapper.unmount();
  });
});
