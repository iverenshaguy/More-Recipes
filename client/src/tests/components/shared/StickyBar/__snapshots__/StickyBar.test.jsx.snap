// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyBarComponent renders correctly 1`] = `
<StickyBar
  currentLocation="view-recipe"
  dispatch={[Function]}
>
  <div
    className="view-recipe"
  >
    <div
      className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none"
    >
      <div
        className="sticky-bar text-center"
      >
        <div
          className="nav-link d-inline"
        >
          <FontAwesome
            className="home-icon"
            name="home"
            onClick={[Function]}
            size="lg"
            tag="i"
          >
            <i
              aria-hidden={true}
              className="fa fa-home fa-lg home-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </div>
        <div
          className="nav-link d-inline"
        >
          <FontAwesome
            className="search-icon"
            name="search"
            onClick={[Function]}
            size="lg"
            tag="i"
          >
            <i
              aria-hidden={true}
              className="fa fa-search fa-lg search-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </div>
        <div
          className="nav-link d-inline"
        >
          <FontAwesome
            id="add-edit-modal-icon"
            name="plus"
            onClick={[Function]}
            size="lg"
            tag="i"
          >
            <i
              aria-hidden={true}
              className="fa fa-plus fa-lg"
              id="add-edit-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </div>
        <div
          className="nav-link d-inline"
        >
          <FontAwesome
            className="favorite-recipes-icon"
            name="heart"
            onClick={[Function]}
            size="lg"
            tag="i"
          >
            <i
              aria-hidden={true}
              className="fa fa-heart fa-lg favorite-recipes-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </div>
        <div
          className="nav-link d-inline"
        >
          <FontAwesome
            className="user-icon"
            name="user"
            onClick={[Function]}
            size="lg"
            tag="i"
          >
            <i
              aria-hidden={true}
              className="fa fa-user fa-lg user-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </div>
      </div>
    </div>
  </div>
</StickyBar>
`;

exports[`StickyBarComponent renders no sticky bar when current location is '/login' 1`] = `
<StickyBar
  currentLocation="/login"
  dispatch={[Function]}
/>
`;

exports[`StickyBarComponent renders no sticky bar when current location is '/signup' 1`] = `
<StickyBar
  currentLocation="/signup"
  dispatch={[Function]}
/>
`;
